src/dataviews/dataview-model-base.js

```diff
@@ -161,15 +174,14 @@ module.exports = Model.extend({
 
   _setupAnalysisStatusEvents: function () {
     this._removeExistingAnalysisBindings();
-    this._analysis = this.getSource();
-    if (this._analysis) {
-      this._analysis.on('change:status', this._onAnalysisStatusChange, this);
+    if (this.getSource()) {
```

Comment:
any special reason for not using `this.has('source')`? looks more descriptive to me

Reply:
I think we should avoid working with attributes as much as possible so that we can change things easier in the future (eg: if we decide `source` is no longer an attribute). If `getSource` is not descriptive enough we can create a `hasSource` method...

Anyway, do we need to check if there's a source? Can't we assume there'll always be one?

Reply:
> do we need to check if there's a source?

IMO is not necessary, but we should update the setters to prevent an empty/invalid source.

thoughts?